src/Formats/ProtobufSerializer.cpp

```diff
@@ -2362,6 +2413,16 @@ namespace
                 missing_columns_filler->addDefaults(mutable_columns, row_num);
         }
 
+        bool nullGoogleWrapper(size_t row_num)
+        {
+            return isGoogleWrapperField(parent_field_descriptor) && mutable_columns[0].get()->isNullAt(row_num);
+        }
+
+        bool nullableGoogleWrapper()
+        {
+            return isGoogleWrapperField(parent_field_descriptor) && mutable_columns[0].get()->isNullable();
+        }
```

Comment:
Names of these functions are bit unclear.

Reply:
It is used only in place, and you can do something like
```
const bool is_google_wrapper = isGoogleWrapperField(parent_field_descriptor);
const bool is_column_nullable = mutable_columns[0].get()->isNullable();
if (is_google_wrapper && is_column_nullable)
  ...
```

Reply:
The trick is that for output I only have access to `mutable_columns` only if it is a Google wrapper (line 2238), so using `&&` in the helper function guards the `mutable_columns[0]` access in case this list is empty.
I can put it back as boolean conditions in the main parts, but these will need something like a nested if or an `&&` operation to prevent second part from firing for non-Google wrappers. I used `&&` earlier and I thought it looked rather ugly.

Reply:
Okay, let's leave it then like it is.